drivers/dai/intel/ssp/ssp.c

```diff
@@ -1586,14 +1687,37 @@ static int dai_ssp_set_config_blob(struct dai_intel_ssp *dp, const struct dai_co
 
 	ssp->state[DAI_DIR_PLAYBACK] = DAI_STATE_PRE_RUNNING;
 	ssp->state[DAI_DIR_CAPTURE] = DAI_STATE_PRE_RUNNING;
+}
 
-	/* ssp blob is set by pcm_hw_params for ipc4 stream, so enable
-	 * mclk and bclk at this time.
-	 */
-	dai_ssp_mn_set_mclk_blob(dp, blob->i2s_driver_config.mclk_config.mdivc,
-				 blob->i2s_driver_config.mclk_config.mdivr);
-	ssp->clk_active |= SSP_CLK_MCLK_ES_REQ;
+static int dai_ssp_set_config_blob(struct dai_intel_ssp *dp, const struct dai_config *cfg,
+				   const void *spec_config)
+{
+	const struct dai_intel_ipc4_ssp_configuration_blob_ver_1_5 *blob15 = spec_config;
+	const struct dai_intel_ipc4_ssp_configuration_blob *blob = spec_config;
+	struct dai_intel_ssp_pdata *ssp = dai_get_drvdata(dp);
+	int err;
 
+	/* set config only once for playback or capture */
+	if (ssp->state[DAI_DIR_PLAYBACK] > DAI_STATE_READY ||
+	    ssp->state[DAI_DIR_CAPTURE] > DAI_STATE_READY)
+		return 0;
+
+	if (blob15->version == SSP_BLOB_VER_1_5) {
+		err = dai_ssp_parse_aux_data(dp, spec_config);
+		if (err)
+			return err;
+		dai_ssp_set_reg_config(dp, cfg, &blob15->i2s_ssp_config);
+		err = dai_ssp_set_clock_control_ver_1_5(dp, &blob15->i2s_clock_control);
+		if (err)
+			return err;
+	} else {
```

Comment:
we need to parse i2s_dma_control in legacy blob, it is supported too.
Checked with the team and actually 1.5 is not used anywhere yet :(

Reply:
so in sof we are not using it for anything. So I would suggest we move to blob1.5 to handle all extra stuff. Having aux data in legacy blob is problematic as it doesn't have size field and the dai api doesn't have size -> we can't tell is there aux data or not. I think the internal spec has said that from tgl we should support blob1.5 and legacy together. Just saying that if it is not used by closed source fw, they should implement it.

Reply:
Both blobs are supported and aux data can be included in both of them.
So not really sure if we can ignore it, especially for windows compatibility.

On linux side topology modification is enough to upgrade blob or we need to update driver too?

Reply:
the aux data support in legacy blob is ambiguous. Spec says there's "optional" i2s_dma_control... and then it should be followed by aux data? how do you know is the optional data there or not, or is it aux data, or both? that's why I would like to move to blob ver1.5. Btw lob ver1.5 should be supported from cavs1.5 upwards, so from apl.

topology changes are enough, no changes to driver.

Reply:
There is a specific TLV ID for i2s_dma_control in spec. This is already implemented and used on closed source so this is not really a huge problem, works in tests and windows. Size needs api modification.
But anyway It would be much easier to use 1.5, just thinking if we can? It depends if windows will make update or stick with legacy forever.

Lets decide about this later and for now marge aux data in 1.5, please just update define above and fix comments in blob

Reply:
yeah for me it was unclear, is the i2s_dma_control part of tlv? spec doesn't say there's tlv, it says there's optional i2s_dma_control. So not sure how it supposed to be detected that legacy blob is followed by optinal i2s_dma_control and then possible optional tlv with aux data. Or is everything inside tlv and if it is, why it doesn't say there's optional tlv.

Anyway, I'll update the ifdef.